### DISCUSSION TITLE
New to fan control experiencing some anomalies

### DISCUSSION THREAD

**Discussion by No-good-deed-goes-unpunished** [Rem0o/FanControl.Releases#1887] [General]
Hello to all from the land downunder :)

I'm new to fan control and to github. Please forgive me.

I've always exclusively used Corsair coolers, power suppliers and Fan controllers (Commander Pro). Occasionally some Corsair RAM.
I wont use iCue. I dsiscovered fan control while on a MSI forum when experiencing random freezes on a PC. It seems the cooler wasn't doing its job. Corsairlink doesn't seem to work on Z790 boards. Have installed the compatability kits for the coolers attached to Z790 boards.

I have downloaded and installed the fancontrol.corsairlink.dll plugin. I had problems getting it detected but eventually fan control discovered it.

ALL case fans are connected to a corsair commander pro
There are three front fans, 1 rear fan. The radiator fans are connected to the pump.
I have set up a graph fan curve and assigned all fans to that curve.
I have selected the CPU Package as the source for monitoring/control
Corsairlink was completely uninstalled before installing Fan Control

The behaviours I'm observing are;
1. I cannot get the rear fan to start at all.
2. One fan is reporting speeds of between 11,000 RPM and 28,000 RPM (Approx)
3. Under controls twol fans are showing nothing for the RPM
4. One fan is showing as connected to the Commander pro and it's fan speed aligns with what HWInfo tell me.
5. The H115iProXT fans controls are not showing any RPM
6. Under speeds fan#1 is showing as 4556 RPM
7. All speeds panels are greyed out.
8. Ive been through the assisted set up a few times but that does not change anything, at least that I can see.
9. I set to start minimized but it not. Also, how to I minimize it back to the system tray?
10. I'm not seeing my PSU on the list. I've select to monitor PSU, and fan is not spinning

I assume I'm "doing it wrong", but I dont know what I dont know.

Would appreciate some guidance
Attached log file and config file. 

[log.txt](https://github.com/Rem0o/FanControl.Releases/files/11954131/log.txt)
[userConfig.zip](https://github.com/Rem0o/FanControl.Releases/files/11954133/userConfig.zip)

System config: 10900X CPU, Corsair H115i Cooler, Corsair HX850i PSU, RTX 2080 Ti GPU, 32GB G.Skill 3600C16Q-32GTZN RAM. Corsair Commander Pro.
![fancontrol](https://github.com/Rem0o/FanControl.Releases/assets/138647241/221ed252-2c48-47c1-9077-9a4f15077aaa)

**Comment by Rem0o** [comment]
Since your whole config revolves around a plugin, I'll link @EvanMulawski here.

There is a lot here, but let's start with the pairing.   
During the assisted setup, the first step is the pairing. Could it pair anything at all? You should be able to see it in the progress dialog.  Something like:

```
Testing Fan 1....
No speed sensors reacted....
```

 From what I can see in your setup, it didn't.

**Reply by EvanMulawski** [reply]
It looks like the one fan reporting the large and wide RPM range is throwing off the automatic pairing process? e.g. If it's changing thousands of RPMs every refresh interval, automatic pairing would see that as responding to the control.

**Reply by Rem0o** [reply]
For sure this will throw it off.  @No-good-deed-goes-unpunished  Are you running that specific fan at the limit where it starts spinning?  If you increase the % 2 or 3 percent, does the rpm gets to reasonable values?

**Comment by EvanMulawski** [comment]
@No-good-deed-goes-unpunished Regarding the Corsair devices:

- Assisted setup should have paired these, but they can be paired manually using the menu on the control ("Pair speed sensor..."). Select the speed sensor with the matching name. Currently, it looks like `Commander PRO Fan #6` is paired with the `Commander PRO Fan #2` sensor.
- PSU support is currently in pre-release. Consult the [device support matrix](https://github.com/EvanMulawski/FanControl.CorsairLink#readme) and go through the footnotes applicable to your device(s).

**Reply by No-good-deed-goes-unpunished** [reply]
Ok, no probs re PSU. I was just a little concerned that its temp was over 55c and the fan wasn't spinning.

I'm not running any fan at any set speed. There are no fans connected to motherboard headers and nothing has been set up in the BIOS. Previously it was managed via CorsairLink, so may be it's running from whatever was last set in CLInk, but that would have been similar to the graph I have set up on FC, not at maximum speed.

Each time I do an assisted set up I get different results as to what is detected and what is not. As it stands now, I have 6 fans (4 case, 2 radiator), and 1 pump. Yet, Fan Control is showing me 8 fan speeds. I gather fan #1 on the motherboard which is reporting 4500 RPM is incorrect?

I have two fans called "Fan #2". One is showing 0 RPM the other is 725RPM. I gather the first sent of speeds are the motherboard fan connectors. They should all be 0 as all fans are on the commander pro. I also gather the second Fan #2 is actually Commander Pro Fan #2 as that's not shown elsewhere.

Right now there are no fans with wildly varying speeds. Just one that is 4500-4600.

I dont understand why in the control section all Command pro cards are greyed out except 1 

If there are no fans connected to any fan header and all are connected to the CPro, why does the controls section show 3 fans sensors active?

**Reply by Rem0o** [reply]
You shouldn't see the cards as fans, but as headers on your motherboard and/or commander.  Each header has 2 "sensors". 1 for control, and 1 that reports rpm, hence the 2 sections. 

Do you have a tiny fan on the VRM or chipset of your motherboard? That might be fan1.

Now, what I would do, turn off every motherboards header controls, knowing nothing is plugged on there. Hide all of these controls.  Now unhide all commander pro controls.  Go into the setting page of the software, and uncheck "show hidden cards". 

Then, turn every control card off (the green toggle).

Now one by one, turn on a commander pro header, turn on manual mode on there (three dot menu on the card), and try to see which fan in your system and rpm sensors reacts to that control, if you got empty channels on your commander, then some controls won't do anything, since they belong to an empty channel.

**Comment by No-good-deed-goes-unpunished** [comment]
Sorry for vanishing. Personal crisis to attend to.
I'm testing the program on 2 PCs.

No, I'm not aware of any chipset fan on the MSI Z790-A pro.
The X299 XPower Gaming AC may indeed have a chipset fan though I can't see it as the mesh is too thick. Doesn't look easy to pull apart either. Perhaps its failed?

The X299 has a commander pro. It has one spare fan header. 
Fan Control was sometimes detecting the Corsair <something> and then sometimes not. It would error at startup.

For The Z790, What I ended up doing was replacing the Corsair cooler with a Deepcool LS520. 

For both PCs I set up fan curves for every fan as even the same brand/model had their individual quirks. And they are maxing out at 20% under max RPM.

I have 10 Pcs and I need to replace 9 coolers due to age and or incompatibility with LGA1700. I plan on running Fan Control on all of them. They all have Corsaur PSUs. This is an excellent piece of software and well worth donating to - which I have. I'm sorry I couldnt afford more.

Perhaps too, the fact that the BIOS was set to Smart fan control with individual controls is contributing to the strange behavior. I just read the FAQ. So I'll now go any fix that.

**Comment by No-good-deed-goes-unpunished** [comment]
PC #1: Test-PC - Turned off smart control of fans in BIOS and then wondered was that applicable since all fans are connected to Commander Pro. This PC picks up all fans plus an extra motherboard fan, which as previously suggested might be a chipset fan. 
This PC has a Corsair HX850i PSU which is picking up 2 sensors called Case and VRM, The case speed card reports 0 RPM and I cannot match and control card to it.
This PC Has a Corsair H115i Pro RGB XT cooler. Fan control picks the pump and fan(s) without issue.

PC#2: Games-PC. This PC has a Deepcool LS520 cooler and an RM750x PSU. This PC is now failing to initialize in an endless loop saying expected 58 sensors. Found 50. I gurss that's related to the fact that I changed coolers and did not disable the corsairlink plugin. RM750x not detected - but as you said it's still experimental.

PC #3: Tanya-PC. This is an X299 system also with 2 commander pro's with 5 fans on each, a HX1200i PSU and a H115i Pro RGB XT cooler. Fan Control does not detect the cooler at all. I was able to match all fans on the CPROs as you suggested above.  After several assisted setups I could not get it to detect the cooler. It detects 2 sensors on the PSU as well - Case and VRM but there I was unable to set up controls for them. There is a speed card called "Case".

I have some visual issues. I cannot see writing on dark mode screens. The fan Control icon is invisible on File Explorer, Start Menu and Task Manager because I'm using light mode and the icon is white.
